lib/llm/src/postprocessor/tool_calling/parsers.rs

```diff
@@ -91,6 +132,30 @@ pub fn try_tool_call_parse(
     }
 }
 
+// Base Detector to call for all tool parsing
+pub fn detect_and_parse_tool_call(
+    message: &str,
+    parser_str: Option<&str>,
+) -> anyhow::Result<Option<ToolCallResponse>> {
+    let mut parser_map: std::collections::HashMap<&str, ToolCallConfig> =
```

Comment:
You don't need to declare the types here, Rust will infer them. It's totally fine to do this if it makes the code clearer though.